Tests/NIOFileSystemFoundationCompatTests/FileSystemFoundationCompatTests.swift

```diff
@@ -33,5 +33,18 @@ final class FileSystemBytesConformanceTests: XCTestCase {
             Date(timeIntervalSince1970: 1.000000001)
         )
     }
+
+    func testReadFileIntoData() async throws {
+        let fs = FileSystem.shared
+        let path = FilePath(#filePath)
+
+        try await fs.withFileHandle(forReadingAndWritingAt: path) { fileHandle in
```

Comment:
Writing into this source file is a _bad_ idea (think about the local development experience of running this test). Reading is fine, writing less so.

Reply:
True, I missed that. I worked with `temporaryFilePath()` locally but swapped it with `FilePath(#filePath)` just before committing. `FilePath(#filePath)` fails for me locally due to permission issues so I was depending on the CI tests and didn't re-test locally after swapping.